src/Sylius/Bundle/ApiBundle/Resources/config/routing/product.yml

```diff
@@ -21,23 +21,23 @@ sylius_api_product_create:
             form: sylius_api_product
 
 sylius_api_product_update:
-    path: /{id}
+    path: /{code}
     methods: [POST, PUT, PATCH]
     defaults:
         _controller: sylius.controller.product:updateAction
         _sylius:
             form: sylius_api_product
```

Comment:
Lets just delete it. It will fix the problem ;) 

@gruberro Change that you have linked is a pretty old and was just a temporary solution. If you met any problem, you are working on some legacy branch of Sylius. Right now, we would like to have as few api form types as possible.